dbms/src/DataTypes/DataTypeFactory.cpp

```diff
@@ -128,6 +115,48 @@ void DataTypeFactory::registerSimpleDataType(const String & name, SimpleCreator
     }, case_sensitiveness);
 }
 
+void DataTypeFactory::registerDataTypeDomain(const String & domain_name, const String & family_name, DataTypeDomainUniquePtr domain, CaseSensitiveness case_sensitiveness)
+{
+    const auto& data_type_creator = findCreatorByName(family_name);
+    all_domains.reserve(all_domains.size() + 1);
+
+    // We can't move the pointer to the Creator since creator is copied later.
```

Comment:
shared_ptr is Ok.

Reply:
Simplified that piece of code, `shared_ptr` is not needed anymore.